nemo/collections/nlp/models/language_modeling/lm_model.py

```diff
@@ -162,64 +159,65 @@ def validation_epoch_end(self, outputs):
         Called at the end of validation to aggregate outputs.
         :param outputs: list of individual outputs of each validation step.
         """
-        avg_loss = torch.stack([x['val_loss'] for x in outputs]).mean()
-        return {'val_loss': avg_loss}
+        if outputs:
+            avg_loss = torch.stack([x['val_loss'] for x in outputs]).mean()
+            return {'val_loss': avg_loss}
 
-    def setup_training_data(self, train_data_layer_config: Optional[Dict]):
-        if 'shuffle' not in train_data_layer_config:
-            train_data_layer_config['shuffle'] = True
+    def setup_training_data(self, cfg: Optional[DictConfig]):
+        if 'shuffle' not in cfg:
```

Comment:
can we add shuffle to train_ds config so we can remove this check?

Reply:
shuffle is already in the train_ds config. This code was there before my port. I'm happy to remove this check though.